pair.py

```diff
@@ -0,0 +1,63 @@
+import itertools
+
+import ortho
+
+
+def projections_work(left_locations, right_locations, left, right, session):
+    for left_location, right_location in zip(left_locations, right_locations):
+        if not (ortho.name_at_location(right, right_location) in session.project_forward(ortho.name_at_location(left, left_location))):
+            return False
+    return True
+
+
+def find_axis_mappings(left, right, session):
+    return [dict(zip(ortho.get_axes(right), permutation)) for permutation in
+            itertools.permutations(ortho.get_axes(left)) if
+            projections_work(permutation, ortho.get_axes(right), left, right, session)]
+
+
+def diagonal_works(left, right):
+    for set_left, set_right in zip(ortho.get_names_as_list_of_diagonal_sets(left)[1:], ortho.get_names_as_list_of_diagonal_sets(right)[:-1]):
+        if not set_left.isdisjoint(set_right):
+            print(f"filtering {left} + {right} due to diagonal rule")
+            return False
+    return True
```

Comment:
Similar refactor into using `all` here. You can make a function out of the predicate that handles printing if you want. (Or, harder mode, a decorator + lambda.)